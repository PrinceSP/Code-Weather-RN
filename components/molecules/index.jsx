import Header from './Header'
import CustomUnits from './CustomUnits'
import Checkbox from './CheckBox'
import HourlyWeather from './HourlyWeather'
import LocationPlace from './LocationPlace'
import SearchHeader from './SearchHeader'

export {Header,CustomUnits,Checkbox,HourlyWeather,LocationPlace,SearchHeader}
