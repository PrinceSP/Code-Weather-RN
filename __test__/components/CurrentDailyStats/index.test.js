import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import {CurrentDailyStats} from '../../../components';

describe('Current Daily Status Component',()=>{
  const componentFunction = jest.fn()
  const dailyWeatherDummy=[
    {
       "dt":1684951200,
       "sunrise":1684926645,
       "sunset":1684977332,
       "moonrise":1684941060,
       "moonset":1684905480,
       "moon_phase":0.16,
       "summary":"Expect a day of partly cloudy with rain",
       "temp":{
          "day":299.03,
          "min":290.69,
          "max":300.35,
          "night":291.45,
          "eve":297.51,
          "morn":292.55
       },
       "feels_like":{
          "day":299.21,
          "night":291.37,
          "eve":297.86,
          "morn":292.87
       },
       "pressure":1016,
       "humidity":59,
       "dew_point":290.48,
       "wind_speed":3.98,
       "wind_deg":76,
       "wind_gust":8.92,
       "weather":[
          {
             "id":500,
             "main":"Rain",
             "description":"light rain",
             "icon":"10d"
          }
       ],
       "clouds":92,
       "pop":0.47,
       "rain":0.15,
       "uvi":9.23
    },
    {
       "dt":1684951200,
       "sunrise":1684926645,
       "sunset":1684977332,
       "moonrise":1684941060,
       "moonset":1684905480,
       "moon_phase":0.16,
       "summary":"Expect a day of partly cloudy with rain",
       "temp":{
          "day":299.03,
          "min":290.69,
          "max":300.35,
          "night":291.45,
          "eve":297.51,
          "morn":292.55
       },
       "feels_like":{
          "day":299.21,
          "night":291.37,
          "eve":297.86,
          "morn":292.87
       },
       "pressure":1016,
       "humidity":59,
       "dew_point":290.48,
       "wind_speed":3.98,
       "wind_deg":76,
       "wind_gust":8.92,
       "weather":[
          {
             "id":500,
             "main":"Rain",
             "description":"light rain",
             "icon":"10d"
          }
       ],
       "clouds":92,
       "pop":0.47,
       "rain":0.15,
       "uvi":9.23
    },
  ]

  it('renders correctly with weather datas',()=>{
    const component = render(
      <CurrentDailyStats dailyDatas={dailyWeatherDummy}/>
    )
    expect(component).toMatchSnapshot()
  })

  it('renders correctly without weather datas',()=>{
    const component = render(
      <CurrentDailyStats dailyDatas={[]}/>
    )
    expect(component).toMatchSnapshot()
  })
})
