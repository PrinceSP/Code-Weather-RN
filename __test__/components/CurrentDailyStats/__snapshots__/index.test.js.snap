// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current Daily Status Component renders correctly with weather datas 1`] = `
<View
  style={
    {
      "marginTop": 20,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#fff",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      Precipitation
    </Text>
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      -
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#fff",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      Probability of precipitation
    </Text>
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      -
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#fff",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      Wind
    </Text>
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      m/s 
       
      <Text />
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#fff",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      Pressure
    </Text>
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#fff",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      Humidity
    </Text>
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#fff",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      UV index
    </Text>
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#fff",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      Sunrise
    </Text>
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      NaN
      :
      NaN
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      Sunset
    </Text>
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      NaN
      :
      NaN
    </Text>
  </View>
</View>
`;

exports[`Current Daily Status Component renders correctly without weather datas 1`] = `
<View
  style={
    {
      "marginTop": 20,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#fff",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      Precipitation
    </Text>
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      -
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#fff",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      Probability of precipitation
    </Text>
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      -
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#fff",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      Wind
    </Text>
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      m/s 
       
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#aaa",
              "fontSize": 18,
            },
            {
              "transform": [
                {
                  "rotate": "undefineddeg",
                },
              ],
            },
            {
              "fontFamily": "FontAwesome6Free-Solid",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {
              "fontWeight": "900",
            },
          ]
        }
      >
        ï„¤
      </Text>
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#fff",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      Pressure
    </Text>
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#fff",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      Humidity
    </Text>
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#fff",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      UV index
    </Text>
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#fff",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      Sunrise
    </Text>
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      NaN
      :
      NaN
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      Sunset
    </Text>
    <Text
      style={
        {
          "color": "#fff",
        }
      }
    >
      NaN
      :
      NaN
    </Text>
  </View>
</View>
`;
